pkg/collector/corechecks/cluster/kubernetesapiserver/kubernetes_openshift_test.go

```diff
@@ -30,9 +30,10 @@ func TestReportClusterQuotas(t *testing.T) {
 	json.Unmarshal(raw, &list)
 	require.Len(t, list.Items, 1)
 
-	prevClusterName := pkgconfigsetup.Datadog().GetString("cluster_name")
-	pkgconfigsetup.Datadog().SetWithoutSource("cluster_name", "test-cluster-name")
-	defer pkgconfigsetup.Datadog().SetWithoutSource("cluster_name", prevClusterName)
+	mockConfig := configmock.New(t)
+	prevClusterName := mockConfig.GetString("cluster_name")
+	mockConfig.SetWithoutSource("cluster_name", "test-cluster-name")
+	defer mockConfig.SetWithoutSource("cluster_name", prevClusterName)
```

Comment:
Is it still needed to save the previous `cluster_name` and to restore it at the end of the test execution?
Shouldn’t the new `test-cluster-name` value of `cluster_name` be scoped to the new `mockConfig`, the lifetime of which is limited to the execution of one test?

Reply:
Removing the defer is a good idea, done.